src/NuGet.Core/NuGet.Commands/RestoreCommand.cs

```diff
@@ -240,9 +240,22 @@ public Task<RestoreResult> ExecuteAsync()
             context.LocalLibraryProviders.Add(
                 new SourceRepositoryDependencyProvider(nugetRepository, _logger, _request.CacheContext));
 
-            foreach (var provider in _request.Sources.Select(s => CreateProviderFromSource(s, _request.CacheContext)))
+            if (_request.SharedCache.RemoteProviders.Any())
```

Comment:
An alternative is to back the RemoteProviders by the cache, so the call to get them either grabs them or returns the cached data rather than the business layer having to know about this condition